src/Tools/dotnet-monitor/Egress/AzureBlobEgressFactory.cs

```diff
@@ -110,14 +112,36 @@ public Provider(AzureBlobEgressProviderOptions options, ILoggerFactory loggerFac
                 IEndpointInfo source,
                 CancellationToken token)
             {
-                // TODO: Add metadata based on source
                 var streamOptions = new AzureBlobEgressStreamOptions();
                 streamOptions.ContentType = contentType;
+                FillBlobMetadata(streamOptions.Metadata, source);
 
                 string blobUri = await _provider.EgressAsync(action, fileName, streamOptions, token);
 
                 return new EgressResult("uri", blobUri);
             }
+
+            private static void FillBlobMetadata(IDictionary<string, string> metadata, IEndpointInfo source)
+            {
+                // Activity metadata
+                metadata.Add(
+                    ActivityMetadataNames.ParentId,
+                    Activity.Current.GetParentId());
```

Comment:
Is Current guaranteed to be non-null, even in Api like situation where the egress provider might be called outside of the context of a web request?

Reply:
I make a change to null check it.

Reply:
Activity can be null

https://github.com/dotnet/aspnetcore/blob/b143a974a5574b9fccedbbe2389aabbfd48fdb6f/src/Hosting/Hosting/src/Internal/HostingApplicationDiagnostics.cs#L48-L55